type/__package_pip/man.rst

```diff
@@ -22,6 +22,10 @@ OPTIONAL PARAMETERS
 name
     If supplied, use the name and not the object id as the package name.
 
+url
```

Comment:
Hmm, do we really need a separate `--url` parameter since we already have `--name`?

I've been installing Pip packages from URLs using:
```sh
pip_pkg=privateproject@git+https://git.example.com/privateproject.git
__package_pip "${pip_pkg%%@*}" --pip pip3 --name "${pip_pkg}"
```

Maybe it makes more sense to deprecate `--name` and change its name to something more expressive?

Reply:
> Hmm, do we really need a separate --url parameter since we already have --name?

Because `--name` is basically alias for object id and imho it's more explicit with `--url`.

Reply:
Having two parameters effectively doing the same thing bothers me.

`--name` is definitely a bad name, because it accepts all sort of values, not just names.

By reading the [source](https://github.com/pypa/pip/blob/22.0.4/src/pip/_internal/commands/install.py#L305) it seems to me that Pip accepts all values that would be acceptable in a `requirements.txt`.

So what if we name the new parameter `--requirement` and deprecate `--name`?

Reply:
if we depend only on object id to grep pip list in explorer/state, is sentence in man enough for that? do we have use case where object id can be something different than `--name`? for `--requirement` it's easy, we just say it can be anything pip supports, but that will not be used in explorer/state...

Reply:
> deprecate `--extra`?

When we have one `--requirement` to rule them all I think it would only be consequent to do so.

> if we depend only on object id to grep pip list in explorer/state, is sentence in man enough for that? do we have use case where object id can be something different than `--name`? for `--requirement` it's easy, we just say it can be anything pip supports, but that will not be used in explorer/state...

Very good point! With `__package` I do occasionally have different `__object_id` and `--name` to support different package names in different package managers.
e.g. I have `__package openssl --name openssl-util` for OpenWrt, but I can `require=__package/openssl` for all OSes.

Without having thought about this too hard I don't think this is a use case for `__package_pip` because it has the same repo on all OSes.

Unless we find some other use cases, I think a sentence in the man page is good enough.

Reply:
pushed, please review